### DISCUSSION TITLE
Batch Labeling Videos in CoLab

### DISCUSSION THREAD

**Discussion by Kagiro-K** [talmolab/sleap#1356] [Help!] [answered]
I had a really basic question to ask as I am a complete novice at coding. I am utilizing your CoLab notebook for the "Predicting and tracking instances in uploaded video" part. I am trying to batch process a large number of clips I've made but was having difficulty writing the correct segment of code for that.
I would love any feedback you can provide!

Here is one example of what I've tried:

```python

import os

# Specify the folder path containing the videos
folder_path = "/content/drive/My Drive/SLEAP/NewSkeleClipped/clips"

# Iterate over the files in the folder
for file_name in os.listdir(folder_path):
    # Construct the full file path
    file_path = os.path.join(folder_path, file_name)
    
    # Check if the item is a file and ends with a video extension
    if os.path.isfile(file_path) and file_name.endswith((".mp4")):
        # Run sleap-track on the video
      !sleap-track {file_path} \
              --frames 0-208\
              --tracking.tracker simple \
              -m models/single_instance_MasterFile
``` 

And this is the error I get afterward:
![image](https://github.com/talmolab/sleap/assets/126974537/3d11cb4e-8471-4661-b5de-4458e717a513)

Would appreciate any guidance on what the error is referring to and how to remedy it.

Thank you!

**Comment by roomrys** [comment, marked as answer]
Hi @Kagiro-K,

SLEAP uses the `from_filename` method from the `Video` class to determine what the backend type should be (i.e. `MediaVideo` for mp4/avi). We should really also print the "specified filename" in the `ValueError`.... My guess would be that your filename might have a space which would then be parsed in as multiple arguments. 

Can you try putting quotes around the `{file_path}` s.t. you call `sleap-track "{file_path}"`. If spaces are the problem, then this will ensure that the spaces do not separate your filepath into multiple arguments.

Can you also add a try, except block around the `sleap-track` command to print the filename (which we should be printing for you in the `ValueError`, but are not at the moment):
```python
try:
    !sleap-track "{file_path}"  # Add the rest of your command
except Exception as e:
    print(f"filepath: {file_path}")
    raise e
```
This will let us know what the `file_path` is (only if the command errors out) and give us some insight into why it is causing problems.

Let us know how it goes.

Thanks,
Liezl

**Comment by Kagiro-K** [comment]
Thank you so much for your response!

Adding the quotations seemed to do the trick! It's unusual because my file names did not have any spaces. But I also moved the trained model into the same folder as the videos, instead of having those files in a subfolder.

This will greatly speed things up in CoLab where I can train, update labels locally, and continue to improve the model.

Thank you!
I really appreciate the speedy response!
________________________________
